blog/2022-07-04-immigrate-erraform-cloud-resources-to-kubeVela.md

```diff
@@ -0,0 +1,136 @@
+
+    ```hcl
+    resource "aws_s3_bucket" "bucket-acl" {
+        bucket = var.bucket
+        acl    = var.acl
+    }
+
+    output "RESOURCE_IDENTIFIER" {
+        description = "The identifier of the resource"
+        value       = aws_s3_bucket.bucket-acl.bucket_domain_name
+    }
+
+    output "BUCKET_NAME" {
+        value       = aws_s3_bucket.bucket-acl.bucket_domain_name
+        description = "The name of the S3 bucket"
+    }
+
+    variable "bucket" {
+        description = "S3 bucket name"
+        default     = "vela-website"
+        type        = string
+    }
+
+    variable "acl" {
+        description = "S3 bucket ACL"
+        default     = "private"
+        type        = string
+    }
+    ```
+
+3. Configure the AWS Cloud provider credentials:
+
+    ```shell
+    export AWS_ACCESS_KEY_ID="your-accesskey-id"
+    export AWS_SECRET_ACCESS_KEY="your-accesskey-secret"
+    export AWS_DEFAULT_REGION="your-region-id"
+    ```
+
+4. Set the variables in the `main.tf` file:
+
+    ```shell
+    export TF_VAR_acl="private"; export TF_VAR_bucket="your-bucket-name"
+    ```
+
+5. (Optional) Create a `backend.tf` to configure your Terraform backend. We just use the default local backend in this example.
+
+6. Run `terraform init` and `terraform apply` to create the S3 bucket:
+
+    ```shell
+    terraform init && terraform apply
+    ```
+
+7. Check the S3 bucket list to make sure the bucket is created successfully.
+
+## Step 2: Get the Terraform State of the Cloud Resource
+
+Run `terraform state pull` to get the Terraform state of the cloud resource and store it into a local file:
+
+```shell
+terraform state pull > state.json
+```
+
+## Step 3: Import Legacy Terraform Cloud Resources into KubeVela
+
+1. Create the `application.yaml` file, please ensure that the description of each field of Component is consistent with your cloud resource configuration:
+
+    ```yaml
+    apiVersion: core.oam.dev/v1beta1
+    kind: Application
+    metadata:
+    name: app-aws-s3
+    spec:
+    components:
+        - name: sample-s3
+        type: aws-s3
+        properties:
+            bucket: vela-website-202110191745
+            acl: private
+
+            writeConnectionSecretToRef:
+            name: s3-conn
+    ```
+
+2. Get the [`backup_restore` tool](https://github.com/kubevela/terraform-controller/tree/master/hack/tool/backup_restore):
+
+    ```shell
+    git clone https://github.com/kubevela/terraform-controller.git
+    cd terraform-controller/hack/tool/backup_restore
+    ```
+
+3. Run the `restore` command:
+    ```shell
+    go run main.go restore --application <path/to/your/application.yaml> --component sample-s3 --state <path/to/your/state.json>
+    ```
+    The above command will resume the Terraform backend in the Kubernetes first and then create the application without recreating the S3 bucket.
+
+That's all! You have successfully immigrate the management of the S3 bucket to KubeVela!
+
```

Comment:
we should describe some use cases of which we can benefit from the migration.

For example, we can update the resources, delete resources with the some lifecycle of the Application.